### PR TITLE
Linting Bazel scripts

### PR DESCRIPTION
**Pull request by guoyuhong**
## What do these changes do?
1. This PR uses [buildifier](https://github.com/bazelbuild/buildtools/tree/master/buildifier) to format the Bazel scripts.
2. The Travis Lint test will check the Bazel script format. A failed Bazel format test will cause Travis test failure, e.g. https://travis-ci.org/guoyuhong/ray/jobs/492612049.
3. Users can `./.travis/bazel-format.sh -f` to format the Bazel scripts.

## Related issue number
https://github.com/ray-project/ray/issues/3986

### CODE REVIEW

**Comment by raulchen** [comment]
Sorry, I should have reported this earlier. I just found bazel has its own [style guide](https://docs.bazel.build/versions/master/skylark/build-style.html), which is slightly different than Python's. And there's a [lint tool](https://docs.bazel.build/versions/master/skylark/skylint.html) for this as well. We'd better use this tool, instead of yapf.

**Comment by guoyuhong** [comment]
@raulchen Skylint is no longer maintained. The next lint file is buildifier.

**Comment by pcmoritz** [comment]
In accordance with the Bazel styleguide, I think there should be spaces around the =, otherwise it looks ugly ;)

**Comment by guoyuhong** [comment]
@pcmoritz Yes, I will apply the buildifier to fix this.

**Comment by guoyuhong** [comment]
@pcmoritz It looks like that `bazel/BUILD.tpl` cannot be formatted when the last two lines `%{PYTHON_INCLUDE_GENRULE}` & `%{PYTHON_IMPORT_LIB_GENRULE}` are present. Do these two lines have alternative ways to use?

**Comment by pcmoritz** [comment]
They are needed and are being substituted in python_configure.bzl, line 259.

One easy way would be to exclude `bazel/BUILD.tpl` from the formatting, can we make that work?

**Comment by guoyuhong** [comment]
@pcmoritz Yes, I formatted the other part of `bazel/BUILD.tpl` and use `find bazel -name "BUILD*" -not -name "*.tpl" -or -name "*.bzl" | xargs` to exclude this file from the lint script.

File: .travis.yml [hunk 1]
```diff
@@ -63,6 +63,16 @@ matrix:
...
         - cd python
         - python setup.py check --restructuredtext --strict --metadata
+        - cd ..
+        # Run Bazel linter Buildifier.
+        - wget https://dl.google.com/go/go1.10.3.linux-amd64.tar.gz
+        - tar -xf go1.10.3.linux-amd64.tar.gz
+        - mkdir $HOME/go_dir
+        - export GOROOT=`pwd`/go
+        - export GOPATH="$HOME/go_dir"
+        - export PATH="$GOROOT/bin:$GOPATH/bin:$PATH"
```
Comment (raulchen) [inline .travis.yml]: I don't think we need to manually install golang. Travis should have pre-installed golang env.  
Here's an example https://gist.github.com/y0ssar1an/df2dab474520c4086926f672c52db139.

Comment (guoyuhong) [reply .travis.yml]: I found pre-installed golang version is 1.7.3, which is not compatible with buildifier.

**Comment by pcmoritz** [comment]
Thanks for doing this! Can we exclude `bazel/python_configure.bzl`, `bazel/cython_library.bzl`, `bazel/BUILD.tpl`, `bazel/BUILD.glog` and `bazel/BUILD.cython`? These are vendored from a different repo with very minimal or no modifications and it will be clearer to keep them as they are (for example if somebody wants to run "diff" to compare with the original versions).

Also, do you know how long it takes to install go in Travis?

**Comment by guoyuhong** [comment]
@pcmoritz 
1. I have reverted the files you mentioned and add a file list into the script instead of a very long command `find bazel -not -name "*.tpl" -not -name "*cython*" -not -name "*glog" -not -name "python_conf*" -not -name "bazel"| xargs`. That is to say if we add a new file we need to add it into the script by hand.
2. From the finished Travis test result, the installation time should be not more than 30 seconds, because the total Lint time is around 3 minutes. https://travis-ci.com/ray-project/ray/builds/100742267

**Review by raulchen — APPROVED** [review]

**Review by pcmoritz — APPROVED** [review]
LGTM!

**Comment by guoyuhong** [comment]
@AmplabJenkins  retest this, please.

**Pull request merged**